features/git-rename-branch/on_main_branch.feature

```diff
@@ -0,0 +1,23 @@
+Feature: git rename-branch: errors with renaming the main branch
+
+  As a developer accidentally trying to rename the main branch
+  I should see an error that this is not possible
+  So that I know that only other branches can be renamed.
+
+
+  Background:
+    Given I am on the "main" branch
+
+
+  Scenario: error when trying to rename
+    When I run `git rename-branch main renamed-main`
+    Then it runs no Git commands
+    And I get the error "The main branch cannot be renamed."
+    And I am still on the "main" branch
+
+
+  Scenario: error when trying to force rename
+    When I run `git rename-branch main renamed-main -f`
+    Then it runs no Git commands
+    And I get the error "The main branch cannot be renamed."
+    And I am still on the "main" branch
```

Comment:
Also, we should specify the behavior for renaming a branch that contains unpushed or unpulled changes. None of these should get lost when renaming a branch, otherwise this command isn't safe.